src/sentry.ts

```diff
@@ -121,10 +122,20 @@ async function getCryptoContext(client: MatrixClient): Promise<CryptoContext> {
     if (!cryptoApi) {
         return {};
     }
-    const keys = [`ed25519:${client.getDeviceEd25519Key()}`];
-    if (client.getDeviceCurve25519Key) {
-        keys.push(`curve25519:${client.getDeviceCurve25519Key()}`);
+
+    let ownDeviceKeys: OwnDeviceKeys | null = null;
+    try {
+        ownDeviceKeys = await cryptoApi.getOwnDeviceKeys();
+    } catch (e) {
+        // Ignore. We'll just not send the keys
+    }
+
+    const keys = [];
+    if (ownDeviceKeys) {
+        keys.push(`ed25519:${ownDeviceKeys.ed25519}`);
+        keys.push(`curve25519:${ownDeviceKeys.curve25519}`);
     }
```

Comment:
Not really sure why `cryptoApi.getOwnDeviceKeys()` would raise an exception unless there was something very broken (in which case, we'd expect `getCrossSigningStatus()` to fail too), so I'd be inclined to get rid of the exception handling here.

But if you want to keep it, I'd simplify it:

```suggestion
    const keys = [];
    try {
        const ownDeviceKeys = await cryptoApi.getOwnDeviceKeys();
        keys.push(`ed25519:${ownDeviceKeys.ed25519}`);
        keys.push(`curve25519:${ownDeviceKeys.curve25519}`);
    } catch (e) {
        // Ignore. We'll just not send the keys
    }
```

Reply:
same in the rageshake code